cave/reader/configurator_run.py

```diff
@@ -40,75 +36,61 @@ def __init__(self, folder: str, ta_exec_dir: Union[str, None]=None):
             in the scenario-object. since instance- and PCS-files are necessary,
             specify the path to the execution-dir of SMAC here
+        file_format: string
+            from [SMAC2, SMAC3, CSV]
         """
-        run_1_existed = os.path.exists('run_1')
-        self.logger = logging.getLogger("cave.SMACrun.{}".format(folder))
-        in_reader = InputReader()
-
+        self.logger = logging.getLogger("cave.ConfiguratorRun.{}".format(folder))
         self.folder = folder
-        self.logger.debug("Loading from %s", folder)
-
-        split_folder = os.path.split(folder)
-        self.logger.info(split_folder)
-        if ta_exec_dir is None:
-            ta_exec_dir = '.'
-
-        self.scen_fn = os.path.join(folder, 'scenario.txt')
-        self.rh_fn = os.path.join(folder, 'runhistory.json')
-        self.validated_rh_fn = os.path.join(folder, 'validated_runhistory.json')
-        self.traj_fn = os.path.join(folder, 'traj_aclib2.json')
-        self.traj_old_fn = os.path.join(folder, 'traj_old.csv')
-
-        # Create Scenario (disable output_dir to avoid cluttering)
-        scen_dict = in_reader.read_scenario_file(self.scen_fn)
-        scen_dict['output_dir'] = ""
-        with changedir(ta_exec_dir):
-            self.scen = Scenario(scen_dict)
-
-        # Load trajectory
-        self.traj = TrajLogger.read_traj_aclib_format(fn=self.traj_fn,
-                                                      cs=self.scen.cs)
-        self.default = self.scen.cs.get_default_configuration()
-        self.incumbent = self.traj[-1]['incumbent']
-        self.train_inst = self.scen.train_insts
-        self.test_inst = self.scen.test_insts
-
-        # Load runhistory
-        self.original_runhistory = RunHistory(average_cost)
-        self.original_runhistory.update_from_json(self.rh_fn, self.scen.cs)
-        if os.path.exists(self.validated_rh_fn):
-            self.validated_runhistory = RunHistory(average_cost)
-            self.validated_runhistory.update_from_json(self.validated_rh_fn, self.scen.cs)
-            self.logger.debug("Found validated runhistory for \"%s\" and using "
-                              "it for evaluation. #configs in validated rh: %d",
-                              self.folder, len(self.validated_runhistory.config_ids))
-            # Check validated runhistory for completeness
-            if not self._check_rh_for_inc_and_def(self.validated_runhistory):
-                self.validated_runhistory = False
+        self.ta_exec_dir = ta_exec_dir
+        self.logger.debug("Loading from \'%s\' with ta_exec_dir \'%s\'.",
+                          folder, ta_exec_dir)
+
+        if file_format == 'SMAC3':
+            self.reader = SMAC3Reader(folder, ta_exec_dir)
+        elif file_format == 'SMAC2':
+            self.reader = SMAC2Reader(folder, ta_exec_dir)
+        elif file_format == 'CSV':
+            self.reader = CSVReader(folder, ta_exec_dir)
```

Comment:
I don't think the CSVReader is imported in the beginning.

Reply:
I didn't implement it yet :D:D thanks for pointing it out, kinda forgot in the smac2-support.